### DISCUSSION TITLE
sgio installation fails

### DISCUSSION THREAD

**Discussion by rhig2000** [wenbinyugroup/.github#29] [General]
Error encountered while installing sgio, shown below. Any suggestions?

> Microsoft Windows [Version 10.0.19045.5011]
> (c) Microsoft Corporation. All rights reserved.
> 
> C:\Codes\VABS\sgio>pip install -r C:\Users\username\Documents\git_projects\sgio\sgio\requirements.txt
> Collecting h5py==3.11.0 (from -r C:\Users\username\Documents\git_projects\sgio\sgio\requirements.txt (line 1))
>   Using cached h5py-3.11.0-cp312-cp312-win_amd64.whl.metadata (2.5 kB)
> Collecting importlib_metadata==8.2.0 (from -r C:\Users\username\Documents\git_projects\sgio\sgio\requirements.txt (line 2))
>   Using cached importlib_metadata-8.2.0-py3-none-any.whl.metadata (4.7 kB)
> Collecting netCDF4==1.7.1.post2 (from -r C:\Users\username\Documents\git_projects\sgio\sgio\requirements.txt (line 3))
>   Using cached netCDF4-1.7.1.post2-cp312-cp312-win_amd64.whl.metadata (1.8 kB)
> Collecting numpy==2.1.1 (from -r C:\Users\username\Documents\git_projects\sgio\sgio\requirements.txt (line 4))
>   Using cached numpy-2.1.1-cp312-cp312-win_amd64.whl.metadata (59 kB)
> Collecting rich==13.8.1 (from -r C:\Users\username\Documents\git_projects\sgio\sgio\requirements.txt (line 5))
>   Using cached rich-13.8.1-py3-none-any.whl.metadata (18 kB)
> Collecting PyYAML==6.0 (from -r C:\Users\username\Documents\git_projects\sgio\sgio\requirements.txt (line 6))
>   Using cached PyYAML-6.0.tar.gz (124 kB)
>   Installing build dependencies ... done
>   Getting requirements to build wheel ... error
>   error: subprocess-exited-with-error
> 
>   × Getting requirements to build wheel did not run successfully.
>   │ exit code: 1
>   ╰─> [54 lines of output]
>       running egg_info
>       writing lib\PyYAML.egg-info\PKG-INFO
>       writing dependency_links to lib\PyYAML.egg-info\dependency_links.txt
>       writing top-level names to lib\PyYAML.egg-info\top_level.txt
>       Traceback (most recent call last):
>         File "C:\Users\username\AppData\Local\Programs\Python\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
>           main()
>         File "C:\Users\username\AppData\Local\Programs\Python\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
>           json_out['return_val'] = hook(**hook_input['kwargs'])
>                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>         File "C:\Users\username\AppData\Local\Programs\Python\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 118, in get_requires_for_build_wheel
>           return hook(config_settings)
>                  ^^^^^^^^^^^^^^^^^^^^^
>         File "C:\Users\username\AppData\Local\Temp\pip-build-env-dbpjicka\overlay\Lib\site-packages\setuptools\build_meta.py", line 333, in get_requires_for_build_wheel
>           return self._get_build_requires(config_settings, requirements=[])
>                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>         File "C:\Users\username\AppData\Local\Temp\pip-build-env-dbpjicka\overlay\Lib\site-packages\setuptools\build_meta.py", line 303, in _get_build_requires
>           self.run_setup()
>         File "C:\Users\username\AppData\Local\Temp\pip-build-env-dbpjicka\overlay\Lib\site-packages\setuptools\build_meta.py", line 319, in run_setup
>           exec(code, locals())
>         File "<string>", line 288, in <module>
>         File "C:\Users\username\AppData\Local\Temp\pip-build-env-dbpjicka\overlay\Lib\site-packages\setuptools\__init__.py", line 117, in setup
>           return distutils.core.setup(**attrs)
>                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>         File "C:\Users\username\AppData\Local\Temp\pip-build-env-dbpjicka\overlay\Lib\site-packages\setuptools\_distutils\core.py", line 183, in setup
>           return run_commands(dist)
>                  ^^^^^^^^^^^^^^^^^^
>         File "C:\Users\username\AppData\Local\Temp\pip-build-env-dbpjicka\overlay\Lib\site-packages\setuptools\_distutils\core.py", line 199, in run_commands
>           dist.run_commands()
>         File "C:\Users\username\AppData\Local\Temp\pip-build-env-dbpjicka\overlay\Lib\site-packages\setuptools\_distutils\dist.py", line 954, in run_commands
>           self.run_command(cmd)
>         File "C:\Users\username\AppData\Local\Temp\pip-build-env-dbpjicka\overlay\Lib\site-packages\setuptools\dist.py", line 999, in run_command
>           super().run_command(command)
>         File "C:\Users\username\AppData\Local\Temp\pip-build-env-dbpjicka\overlay\Lib\site-packages\setuptools\_distutils\dist.py", line 973, in run_command
>           cmd_obj.run()
>         File "C:\Users\username\AppData\Local\Temp\pip-build-env-dbpjicka\overlay\Lib\site-packages\setuptools\command\egg_info.py", line 312, in run
>           self.find_sources()
>         File "C:\Users\username\AppData\Local\Temp\pip-build-env-dbpjicka\overlay\Lib\site-packages\setuptools\command\egg_info.py", line 320, in find_sources
>           mm.run()
>         File "C:\Users\username\AppData\Local\Temp\pip-build-env-dbpjicka\overlay\Lib\site-packages\setuptools\command\egg_info.py", line 541, in run
>           self.add_defaults()
>         File "C:\Users\username\AppData\Local\Temp\pip-build-env-dbpjicka\overlay\Lib\site-packages\setuptools\command\egg_info.py", line 579, in add_defaults
>           sdist.add_defaults(self)
>         File "C:\Users\username\AppData\Local\Temp\pip-build-env-dbpjicka\overlay\Lib\site-packages\setuptools\command\sdist.py", line 109, in add_defaults
>           super().add_defaults()
>         File "C:\Users\username\AppData\Local\Temp\pip-build-env-dbpjicka\overlay\Lib\site-packages\setuptools\_distutils\command\sdist.py", line 238, in add_defaults
>           self._add_defaults_ext()
>         File "C:\Users\username\AppData\Local\Temp\pip-build-env-dbpjicka\overlay\Lib\site-packages\setuptools\_distutils\command\sdist.py", line 323, in _add_defaults_ext
>           self.filelist.extend(build_ext.get_source_files())
>                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>         File "<string>", line 204, in get_source_files
>         File "C:\Users\username\AppData\Local\Temp\pip-build-env-dbpjicka\overlay\Lib\site-packages\setuptools\_distutils\cmd.py", line 107, in __getattr__
>           raise AttributeError(attr)
>       AttributeError: cython_sources
>       [end of output]
> 
>   note: This error originates from a subprocess, and is likely not a problem with pip.
> error: subprocess-exited-with-error
> 
> × Getting requirements to build wheel did not run successfully.
> │ exit code: 1
> ╰─> See above for output.
> 
> note: This error originates from a subprocess, and is likely not a problem with pip.
>

**Comment by unitasium** [comment]
Can you try the following configuration:

Use python 3.9 and install the following required packages:
```
h5py
importlib_metadata
netCDF4
numpy<2
rich
PyYAML
```
The key is to use numpy version less than 2.0.
